pkg/controller/jobs/pod/pod_webhook.go

```diff
@@ -77,9 +87,89 @@ func SetupWebhook(mgr ctrl.Manager, opts ...jobframework.Option) error {
 
 var _ webhook.CustomDefaulter = &PodWebhook{}
 
+func omitKueueLabels(l map[string]string) map[string]string {
+	result := map[string]string{}
+
+	for key, value := range l {
+		if !strings.HasPrefix(key, "kueue.x-k8s.io/") {
+			result[key] = value
+		}
+	}
+	return result
+}
+
+func containersShape(containers []corev1.Container) (result []map[string]interface{}) {
+	for _, c := range containers {
+		result = append(result, map[string]interface{}{
+			"image": c.Image,
+			"resources": map[string]interface{}{
+				"requests": c.Resources.Requests,
+			},
+			"ports": c.Ports,
+		})
+	}
+
+	return result
+}
+
+func volumesShape(volumes []corev1.Volume) (result []corev1.Volume) {
+	for _, v := range volumes {
+		v.Name = ""
+		result = append(result, v)
+	}
+
+	return result
+}
+
+func getRoleHash(p *Pod) (string, error) {
+
+	shape := map[string]interface{}{
+		"metadata": map[string]interface{}{
+			"labels": omitKueueLabels(p.pod.ObjectMeta.Labels),
+		},
+		"spec": map[string]interface{}{
+			"initContainers":            containersShape(p.pod.Spec.InitContainers),
+			"containers":                containersShape(p.pod.Spec.Containers),
+			"nodeSelector":              p.pod.Spec.NodeSelector,
+			"affinity":                  p.pod.Spec.Affinity,
+			"tolerations":               p.pod.Spec.Tolerations,
+			"runtimeClassName":          p.pod.Spec.RuntimeClassName,
+			"priority":                  p.pod.Spec.Priority,
+			"preemptionPolicy":          p.pod.Spec.PreemptionPolicy,
+			"topologySpreadConstraints": p.pod.Spec.TopologySpreadConstraints,
+			"overhead":                  p.pod.Spec.Overhead,
+			"volumes":                   volumesShape(p.pod.Spec.Volumes),
```

Comment:
oh cool. We might need to do the same for resourceClaims, but let's leave that for a follow up.